### PR TITLE
Regular readable time

### PR DESCRIPTION
**Pull request by idodeclare**
Hello,

Please consider for integration this patch so `PendingFileCompleter` uses the regular OpenGrok `getReadableTime()` instead of `DurationFormatUtils` so that sub-second detail is provided.

`StringUtils.getReadableTime()` is updated to use `StringBuilder` and to avoid some pointless high-precision (e.g., "1 day 35 ms") when for other intervals milliseconds are not shown. It is also updated not just to report "0" for near-instant intervals but "0 ms".

Thank you.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/util/StringUtils.java [hunk 1]
```diff
@@ -116,53 +116,61 @@ public static boolean isOnlyWhitespace(String str) {
...
-    //    - class must be qualified with a package name
-    //    - only letters in package name, starting with lower case
-    //    - class name must be in CamelCase, starting with upper case
-    return javaClassPattern.matcher(s).matches();
-  }
+        // Only match a small subset of possible class names to prevent false
+        // positives:
+        //    - class must be qualified with a package name
+        //    - only letters in package name, starting with lower case
+        //    - class name must be in CamelCase, starting with upper case
```
Comment (tulinkry) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/util/StringUtils.java:123]: consider moving this comment to the javadoc as it's valuable for the caller

File: opengrok-indexer/src/test/java/org/opengrok/indexer/util/StringUtilsTest.java [hunk 2]
```diff
@@ -39,12 +39,12 @@ public void testValues() {
...
-            1075634299
+            1075634299, 86480001
         };
         String[] expected = {
-            "0", "100 ms", "1.0 seconds", "1.500 seconds", "0:01:04",
-            "0:02:04", "1:00:51", "1 day", "1 day 34.349 seconds",
-            "12 days 10:47:14"
+            "0 ms", "100 ms", "1.0 seconds", "1.500 seconds", "0:01:04",
+            "0:02:04", "1:00:51", "1 day", "1 day",
+            "12 days 10:47:14", "1 day 00:01:20"
```
Comment (tulinkry) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/util/StringUtilsTest.java:47]: could you add a `86400001` (one day + one ms) too?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/util/StringUtils.java (hunk 1 above)
Comment (idodeclare) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/util/StringUtils.java]: OK done

File: opengrok-indexer/src/test/java/org/opengrok/indexer/util/StringUtilsTest.java (hunk 2 above)
Comment (idodeclare) [reply opengrok-indexer/src/test/java/org/opengrok/indexer/util/StringUtilsTest.java]: OK

**Comment by idodeclare** [comment]
Approved by @tulinkry on 6 May; what do you require for a merge?

**Comment by tarzanek** [comment]
from my point of view only the travis mac build failed, let me rerun, if it will be green, me or Vlada can merge (I'd still love Oracle to give back the repos, so we can add more people with merge permissions)

**Comment by tarzanek** [comment]
ok, I've rerun the job, but I doubt it has anything to do with your changes (was a download error)

**Comment by vladak** [comment]
The last Travis build seems to be okay, merging.

**Comment by vladak** [comment]
Thanks !

**Comment by idodeclare** [comment]
Thank you, @tarzanek and @vladak !

**Pull request merged**